File: app/src/main/java/com/song/deviceinfo/info/HardwareInfo.java
package com.song.deviceinfo.info;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorManager;

import java.util.ArrayList;
import java.util.List;

import androidx.core.util.Pair;

import static android.content.Context.SENSOR_SERVICE;

/**
 * Created by chensongsong on 2020/9/22.
 */
public class HardwareInfo {

    public static List<Pair<String, String>> getHardwareInfo(Context context) {
        List<Pair<String, String>> list = new ArrayList<>();
        SensorManager sensorManager = (SensorManager) context.getSystemService(SENSOR_SERVICE);
        Sensor gyroscope = sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE);
        list.add(new Pair<>("GYROSCOPE", gyroscope.getName()));
        Sensor magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
        list.add(new Pair<>("MAGNETIC", magnetic.getName()));
        Sensor accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
        list.add(new Pair<>("ACCELEROMETER", accelerometer.getName()));
        return list;
    }


}

File: app/src/main/java/com/song/deviceinfo/info/HardwareInfo.kt
package com.song.deviceinfo.info

import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorManager
import androidx.core.util.Pair

/**
 * Created by chensongsong on 2020/9/22.
 */
object HardwareInfo {
    fun getHardwareInfo(context: Context): List<Pair<String, String>> {
        val list: MutableList<Pair<String, String>> = ArrayList()
        val sensorManager = context.getSystemService(Context.SENSOR_SERVICE) as SensorManager
        val gyroscope = sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE)
        list.add(Pair("GYROSCOPE", gyroscope!!.name))
        val magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD)
        list.add(Pair("MAGNETIC", magnetic!!.name))
        val accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
        list.add(Pair("ACCELEROMETER", accelerometer!!.name))
        return list
    }
}
